### PR TITLE
Feature/vehicle detail view

### LINKED ISSUE #26: Vehicle/Location data not loaded for breacrumbs
When loading or refreshing a page with a vehicle or location, the breadcrumbs sometimes don't show the vehicle VIN or location name because that data has not been loaded.

FleetView.js uses data from the redux store to display vehicle VINs and location names in the breadcrumbs component, but this data might not be in the store if the page was just loaded or refreshed.

screen shot 2017-10-11 at 2 12 08 pm

### PR DESCRIPTION
**Pull request by ryanstrandt**
fixes #26 along with #36

### CODE REVIEW

File: src/components/fleet/VehicleDetail.js [hunk 1]
```diff
@@ -1,16 +1,141 @@
...
+  componentDidMount() {
+    const { action, match } = this.props;
+    action.getVehicle(match.params.vehicleId);
+  }
+
+  render() {
+    const { match, vehicleData } = this.props;
+    return (
+      <div>
+        <h3>{`Vehicle Detail: ${vehicleData.vehicle.vin}`}</h3>
```
Comment (gwenf) [inline src/components/fleet/VehicleDetail.js]: When I navigate to `http://localhost:8080/fleet/locations/:locationId/vehicles/:vehicleId` through the breadcrumbs this throws an error. 
screen shot 2018-01-01 at 9 36 18 am

Comment (ryanstrandt) [reply src/components/fleet/VehicleDetail.js]: I just branched straight from development and added this view and I can't get the error. How did you get there. What did you click on.

File: src/components/common/form/dropdownField.js [hunk 2]
```diff
@@ -2,20 +2,20 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { FormControl } from 'react-bootstrap';
 
-const DropDownSelect = ({ input, states }) => {
+const DropDownSelect = ({ input }) => {
 
-  function renderSelectOptions (state) {
-    return <option key={state.stateCode} value={state.stateCode}>{state.name}</option>;
-  }
+  // function renderSelectOptions (state) {
```
Comment (BWortman) [inline src/components/common/form/dropdownField.js]: Please don't check in commented out code.

File: src/components/fleet/VehicleDetail.js (hunk 1 above)
Comment (BWortman) [reply src/components/fleet/VehicleDetail.js]: I got the error, too. From the Locations page, click Vehicles in the Actions column (any row). Then click Edit (any row). Now click the VIN in the breadcrumbs to get the error.

**Comment by BWortman** [comment]
Added @camoore1979  to help with this, now that @ryanstrandt  is no longer as available (no longer on the bench).

**Comment by camoore1979** [comment]
@BernzSed @BWortman fixed the bug above. Abstracted VehicleDetail into a Container and a dumb component.

**Comment by BWortman** [comment]
Thanks, @camoore1979. Looks pretty good, but there's been a regression. It crashes when trying to display the Location view.

**Comment by camoore1979** [comment]
@BWortman I'm not seeing that behavior locally. Let's be sure we're looking at the same place though, so I can confirm. 

I'm able to visit all the following routes locally with no problem on this branch:

- http://localhost:8080/fleet/locations/
-  http://localhost:8080/fleet/locations/c0b694ec-3352-43e3-9f22-77c87fe83d48
-  http://localhost:8080/fleet/locations/c0b694ec-3352-43e3-9f22-77c87fe83d48/edit
- http://localhost:8080/fleet/locations/c0b694ec-3352-43e3-9f22-77c87fe83d48/vehicles

I double checked that I've pushed up all my commits, so this is the latest code.

**Comment by BWortman** [comment]
The second path you listed was the one that crashed. Does this happen to anyone else?

Sent from my Windows 10 phone

________________________________

**Comment by camoore1979** [comment]
@BWortman I assume you did a fresh `git pull` on that branch! ;-)

**Comment by BWortman** [comment]
Yes, @camoore1979. I did a `git clean -dxf`, `npm install`, `npm test`, and then `npm start`. Logged in, and clicked the Site ID:

image

I get the error every time, regardless of which row:

image
I also get it when I click the breadcrumb:
image
This happens every time, so it's consistent. Shouldn't it be accessing the location by siteId instead of name?

**Comment by camoore1979** [comment]
@BWortman Oh the shame!!! (hangs head). It was a local caching issue for me. I am now recreating the problem. Also, will solve the caching issue with a webpack config change (one line fix!). And will fix da bug!

**Comment by BWortman** [comment]
Thanks, Chad!


________________________________

**Comment by BWortman** [comment]
Hi @camoore1979  - now there's a test failing, at least for me. Did you run `npm test`?

**Comment by camoore1979** [comment]
OK. The intended functionality for this story / issue should now be working!

- resolved the bug with the location detail view
- updated the NavBar snapshot for the NavBar unit test.
- updated the Webpack config to add a hash to the js bundled assets, for cache busting

You should be able to run the app and reach these routes successfully:
- location detail: http://localhost:8080/fleet/locations/c0b694ec-3352-43e3-9f22-77c87fe83d48
- vehicles list: http://localhost:8080/fleet/locations/c0b694ec-3352-43e3-9f22-77c87fe83d48/vehicles
- vehicle detail: http://localhost:8080/fleet/locations/c0b694ec-3352-43e3-9f22-77c87fe83d48/vehicles/0b68a1d6-4210-42e0-9647-3747da20a181

@BWortman, @gwenf, @ryanstrandt please take a look! Thank you!

**Comment by BWortman** [comment]
@BernzSed - Can you take a quick look at the code, just to look for anything major that I overlooked?

**Comment by BWortman** [comment]
I'm going to go ahead and merge this. The Issue still isn't closed b/c the screen isn't working per the spec, but this is a step forward nevertheless.

**Pull request merged**